tools/walletextension/wallet_extension.go

```diff
@@ -346,3 +352,45 @@ func adjustStateRoot(rpcResp interface{}, respMap map[string]interface{}) {
 		}
 	}
 }
+
+// handleSpecialRPCMethods checks method name in the request and if it is one of the methods that we use differently than
+// eth standard. At the moment it is only getStorageAt method. It returns response if we should use that response or nil otherwise
+func (w *WalletExtension) handleSpecialRPCMethods(request *accountmanager.RPCRequest, hexUserID string) map[string]interface{} {
+	response := map[string]interface{}{}
+	// all responses must contain the request id. Both successful and unsuccessful.
+	response[common.JSONKeyRPCVersion] = jsonrpc.Version
+	response[common.JSONKeyID] = request.ID
+
+	// check if requested method is GetStorageAt - as we use it for different purpose here with specific parameters
+	if request.Method == rpc.GetStorageAt && validateGetStorageAtForUserIDParams(request.Params) {
+		// check if userID in the parameters is also in our database
+		userID, err := common.GetUserIDbyte(hexUserID)
+		if err != nil {
+			return nil
+		}
+
+		key, err := w.storage.GetUserPrivateKey(userID)
+		if err != nil || len(key) == 0 {
+			w.logger.Info("Trying to get userID, but it is not present in our database: ")
+			return nil
+		}
+
+		response[common.JSONKeyResult] = hexUserID
+		return response
+	}
+
+	return nil
+}
+
+func validateGetStorageAtForUserIDParams(params []interface{}) bool {
+	for _, item := range params {
+		if m, ok := item.(map[string]interface{}); ok {
+			address, addrOk := m["address"].(string)
+			position, posOk := m["position"].(string)
+			if addrOk && address == "obscuro" && posOk && position == "0" {
```

Comment:
These should be constants. In case we'll add more methods like this, we'll create an enum or something more generic

Reply:
fixed, thanks.